### PR TITLE
added TrainingModelWithProgress and examples

### PR DESCRIPTION
**Pull request by bytesofdhiren**

### CODE REVIEW

File: examples/GetTrainingModelProgress.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public static void main(String args[]) throws IOException {
+        IndicoConfig config = new IndicoConfig.Builder()
+                .host("dev.indico.io")
+                .tokenPath("__TOKEN_PATH__")
+                .build();
+
+        try (IndicoClient client = new IndicoClient(config)) {
+            ModelGroupQuery modelGroupQuery = client.modelGroupQuery();
+            TrainingModelWithProgressQuery trainingModelWithProgress = client.trainingModelWithProgressQuery();
+            ModelGroup modelGroup = modelGroupQuery.id(__MODEL_ID__).query();
```
Comment (nkassis) [inline examples/GetTrainingModelProgress.java:20]: Call these __MODEL_GROUP_ID__ as model id will be confuse with selected model inside a model group.

File: src/main/java/com/indico/query/TrainingModelWithProgressQuery.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        TrainingProgress progress = new TrainingProgress(model.trainingProgress().percentComplete());
+        return new Model.Builder()
+                .id(model.id())
+                .status(model.status())
+                .trainingProgress(progress)
+                .build();
+    }
+
+    @Override
+    public Model refresh(Model obj) {
```
Comment (nkassis) [inline src/main/java/com/indico/query/TrainingModelWithProgressQuery.java:73]: we should drop this

Comment (bytesofdhiren) [reply src/main/java/com/indico/query/TrainingModelWithProgressQuery.java:73]: refresh is part of Query interface, do you want to drop it for all queries?

File: examples/GetTrainingModelProgress.java (hunk 1 above)
Comment (bytesofdhiren) [reply examples/GetTrainingModelProgress.java]: Fixed.

File: src/main/java/com/indico/request/GraphQLRequest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    public GraphQLRequest(IndicoClient client) {
+        this.client = client;
+    }
+
+    public GraphQLRequest query(String query) {
+        this.query = query;
+        return this;
+    }
+
+    public GraphQLRequest operationName(String operationName) {
```
Comment (nkassis) [inline src/main/java/com/indico/request/GraphQLRequest.java:30]: Operation name can be removed it doesn't do anything sent separately.

File: src/main/java/com/indico/request/GraphQLRequest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    public GraphQLRequest variables(JSONObject variables) {
+        this.variables = variables;
+        return this;
+    }
+
+    @Override
+    public JSONObject call() throws IOException {
+        JSONObject json = new JSONObject();
+        json.put("query", this.query);
+        json.put("operationName", this.operationName);
```
Comment (nkassis) [inline src/main/java/com/indico/request/GraphQLRequest.java:44]: remove

File: examples/GraphQL.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                    + "        status\n"
+                    + "        rowCount\n"
+                    + "        numModelGroups\n"
+                    + "        modelGroups {\n"
+                    + "            id\n"
+                    + "        }\n"
+                    + "    }\n"
+                    + "}";
+
+            JSONObject response = request.query(query).operationName("GetDatasets").call();
```
Comment (nkassis) [inline examples/GraphQL.java:31]: operationName can be removed here it's not used for anything.

Comment (bytesofdhiren) [reply examples/GraphQL.java]: Fixed. Made Optional.

File: src/main/java/com/indico/request/GraphQLRequest.java (hunk 4 above)
Comment (bytesofdhiren) [reply src/main/java/com/indico/request/GraphQLRequest.java]: Fixed

File: src/main/java/com/indico/request/GraphQLRequest.java (hunk 3 above)
Comment (bytesofdhiren) [reply src/main/java/com/indico/request/GraphQLRequest.java:30]: Fixed.

**Pull request merged**